pymeasure/instruments/apex/apex1000.py

```diff
@@ -0,0 +1,176 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2020 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+from pymeasure.instruments import Instrument, discreteTruncate
+from pymeasure.instruments.validators import strict_discrete_set, \
+    truncated_discrete_set, truncated_range
+
+import numpy as np
+import time
+import re
+
+
+class ApexAP1000(Instrument):
+    """Represents Apex's AP1000 Multi-test platform"""
+    SLOTS = range(1,9)
+    DEV_KINDS = {'PM':'POW','TF':'FLT','TL':'TLS'}
+    PM_CHANNELS = [1,2]
+    PM_UNIT = ['W','mW','dBm']
+    STATUS = {'Off':0,'On':1}
+
+
+    def __init__(self, resourceName, **kwargs):
+        super(ApexAP1000, self).__init__(
+            resourceName,
+            "Apex AP1000",
+            **kwargs
+        )
+        self.__slot = None
+
+    def headStr(self,kind):
+        return ApexAP1000.DEV_KINDS[kind]+"[%02d]:"%self.slot
+
+    @property
+    def slot(self):
+        """ Get current slot number
+        """
+        if self.__slot in ApexAP1000.SLOTS:
+            return self.__slot
+        else:
+            raise ValueError('Bad slot number !')
```

Comment:
For debugging purposes, I would suggest adding an `elif self.__slot is None` case to check whether the slot has been set at al.